layers/+lang/go/funcs.el

```diff
@@ -117,3 +117,15 @@
 (defun spacemacs/go-packages-gopkgs ()
   "Return a list of all Go packages, using `gopkgs'."
   (sort (process-lines "gopkgs") #'string<))
+
+(defun spacemacs/go-install-after-save ()
+  "Install current package to pkg."
+  (interactive)
+  (when (eq major-mode 'go-mode)
+    (let ((errbuf (get-buffer-create "*GoInstall*"))
+          (install-args (list "install" "--buildmode=archive")))
+      (if (eq go-install-after-save 'sync)
+          (if (zerop (apply #'call-process "go" nil errbuf nil install-args))
+              (message "Applied go install")
+            (message "Apply go install failed")))
+      (apply #'start-process "go-install" errbuf "go" install-args))))
```

Comment:
Shouldn't this `apply` be inside the `if` above? As I understand it setting the `go-install-after-save` to `sync` will execute the install command once synchronously and the second time asynchronously.

Reply:
no, it is not necessary to run install twice, just choose sync or async

Reply:
Then `(apply #'start-process...` needs to be inside the `if`. Also instead of an if I would like to have a `cond` that checks for `eq go-install-after-save 'sync` and `eq go-install-after-save t` (or `async`) since there are 3 possible values for that variable, `nil` included.

Reply:
I got it, you mean this if: 
```
      (if (eq go-install-after-save 'sync)
```
fixed.

and about the checks:
I think it should check nil before call this function
so, when called, it will do async as default, or sync as configured.
I don't know if double check is conversions, should I check nil here?

Reply:
I guess since the code is simple enough then the second nil check is not needed. It should be OK as it is right now.